File: src/main/java/com/example/project_test/Entities/Question.java
package com.example.project_test.Entities;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Question implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long idQuestion;
    String descriptionQuestion;
    @JsonManagedReference
    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL)
    private List<Reponse> reponses;
    LocalDateTime dateAjoutQ;


    @PrePersist
    private void setDateAjoutQ() {
        this.dateAjoutQ = LocalDateTime.now();
    }
}

File: src/app/model/question.ts
import {Reponse} from "./reponse";
import {Etudiant} from "./etudiant";

export class Question{
  idQuestion!: number ;
  descriptionQuestion!: string;
  reponses!: Reponse[];
  dateAjoutQ!: Date;
  idEtudiant!: Etudiant;



}
